kani-compiler/src/codegen_aeneas_llbc/mir_to_ullbc/mod.rs

```diff
@@ -624,14 +644,23 @@ impl<'a, 'tcx> Context<'a, 'tcx> {
         self.translate_span(instance.def.span())
     }
 
+    fn file_to_id(&mut self, filename: &CharonFileName) -> Option<CharonFileId> {
+        for (id, file) in self.translated.files.iter().enumerate() {
```

Comment:
Iterating over all files every time we translate a span could be expensive. We should introduce a hash table of file to id in `Context` and avoid relying on the `translated.files` one. Charon does that as well: https://github.com/AeneasVerif/charon/blob/0de54092afb546bf53cd8261c79499f3cae2c24b/charon/src/bin/charon-driver/translate/translate_ctx.rs#L179